### ISSUE TITLE
autoexplore stops without printing a message beside runed door vault

### ISSUE THREAD

**Issue by falsifian** [crawl/crawl#1597]
**PLATFORM and OS or BROWSER:**
webtiles; Firefox 8.10 (OpenBSD)

**DCSS VERSION:**
0.25.1-2-g60ff931df9 (on cbro.berotato.org)

**DESCRIPTION:**
In the following situation, when I press "o", nothing happens (no message is printed).

Also, if I move one square up-left and press "o", autoexplore just takes me back exactly here and stops.

I thought autoexplore should ignore the vault, since it's completely closed off. If there is some reason it's stopping, autoexplore should print a message explaining why.

![image](https://user-images.githubusercontent.com/225893/97191290-a4145680-179e-11eb-84ec-6c4d315a3d1a.png)

**EXTRA FILES:**
[Here's a backup file](https://cbro.berotato.org//falsifian-crawl-0.25-201026-1523.tar.bz2).

**Comment by falsifian** [comment]
Here's my rc file (copied and pasted from the web UI). I think the `view_max_width` and `view_max_height` at the top are the only changes I made.

```
view_max_width=81
view_max_height=71

##### Crawl Init file ###############################################
# For descriptions of all options, as well as some more in-depth information
# on setting them, consult the file
#    options_guide.txt
# in your /docs directory. If you can't find it, the file is also available
# online at:
# https://github.com/crawl/crawl/blob/master/crawl-ref/docs/options_guide.txt
#
# Crawl uses the first file of the following list as its option file:
#  * init.txt in the -rcdir directory (if specified)
#  * .crawlrc in the -rcdir directory (if specified)
#  * init.txt (in the Crawl directory)
#  * ~/.crawl/init.txt (Unix only)
#  * ~/.crawlrc (Unix only)
#  * ~/init.txt (Unix only)
#  * settings/init.txt (in the Crawl directory)

##### Some basic explanation of option syntax #######################
# Lines beginning with '#' are comments. The basic syntax is:
#
# field = value         or      field.subfield = value
#
# Only one specification is allowed per line.
#
# The terms are typically case-insensitive except in the fairly obvious
# cases (the character's name and specifying files or directories when
# on a system that has case-sensitive filenames).
#
# White space is stripped from the beginning and end of the line, as
# well as immediately before and after the '='. If the option allows
# multiple comma/semicolon-separated terms (such as
# autopickup_exceptions), all whitespace around the separator is also
# trimmed. All other whitespace is left intact.
#
# There are three broad types of Crawl options: true/false values (booleans),
# arbitrary values, and lists of values. The first two types use only the
# simple =, with later options - which includes your options that are different
# from the defaults - overriding earlier ones. List options allow using +=, ^=,
# -=, and = to append, prepend, remove, and reset, respectively. Usually you will
# want to use += to add to a list option. Lastly, there is := which you can use
# to create an alias, like so:
#   ae := autopickup_exceptions
# From there on, 'ae' will be treated as if it you typed autopickup_exceptions,
# so you can save time typing it.
#

##### Other files ###################################################
# You can include other files from your options file using the 'include'
# option. Crawl will treat it as if you copied the whole text of that file
# into your options file in that spot. You can uncomment some of the following
# lines by removing the beginning '#' to include some of the other files in
# this folder.

# Some useful, more advanced options, implemented in LUA.
# include = advanced_optioneering.txt

# Alternative vi bindings for Dvorak users.
# include = dvorak_command_keys.txt

# Alternative vi bindings for Colemak users.
# include = colemak_command_keys.txt

# Alternative vi bindings for Neo users.
# include = neo_command_keys.txt

# Override the vi movement keys with a non-command.
# include = no_vi_command_keys.txt

# Turn the shift-vi keys into safe move, instead of run.
# include = safe_move_shift.txt

##### Ancient versions ##############################################
# If you're used to the interface of ancient versions of Crawl, you may
# get back parts of it by uncommenting the following options:

# include                 = 034_command_keys.txt

# And to revert monster glyph and colouring changes:

# include                 = 052_monster_glyphs.txt
# include                 = 060_monster_glyphs.txt
# include                 = 071_monster_glyphs.txt
# include                 = 080_monster_glyphs.txt
# include                 = 0.9_monster_glyphs.txt
# include                 = 0.12_monster_glyphs.txt
# include                 = 0.13_monster_glyphs.txt
# include                 = 0.14_monster_glyphs.txt
```

**Comment by damerell** [comment]
This is the behaviour with any ordinary closed door when "travel_open_doors = false" - but even then it normally says something when a runed door remains. (I regret I've just come to internalise it giving no feedback when autoexplore wants to go through a closed door).

**Comment by Aliscans** [comment]
When explore is stopped because of a runed door, whether you get a message depends on whether you've seen the square(s) just behind the door.

If you haven't, _explore_find_target_square() stops you from exploring, and gives a message. 

If you have, _explore_find_target_square() returns one of the squares inside the vault, and find_travel_pos() stops the exploration once you reach the runed door. find_travel_pos() doesn't give a message in this situation, as you've seen.

I may have got one or two details wrong here, but that's the general idea.

**Comment by falsifian** [comment]
Hm, for some reason I thought autoexplore just completely ignored things behind transparent runed doors. Does autoexplore specifically ignore ghost vaults? If so that might explain my misconception; I kind of lump everything involving transparent runed doors together in my mind.

**Comment by ebering** [comment]
@Aliscans door patch, which was merged in 0.26, resolves this issue and now a message is printed if autoexplore cannot continue because of a runed door.
